src/Components/test/E2ETest/ServerExecutionTests/PrerenderingTest.cs

```diff
@@ -72,6 +72,51 @@ public void IsCompatibleWithLazyLoadWebAssembly()
             AssertLogDoesNotContainCriticalMessages("Could not load file or assembly 'Newtonsoft.Json");
         }
 
+        [Fact]
+        public void CanInfluenceHeadDuringPrerender()
+        {
+            Navigate("/prerendered/prerendered-head");
+
+            var metaWithBindings = Browser.FindElement(By.Id("meta-with-bindings"));
+            var metaNoBindings = Browser.FindElement(By.Id("meta-no-bindings"));
+
+            // Validate updated head during prerender
+            Browser.Equal("Initial title", () => Browser.Title);
+            Browser.Equal("Initial meta content", () => metaWithBindings.GetAttribute("content"));
+            Browser.Equal("Immutable meta content", () => metaNoBindings.GetAttribute("content"));
+
+            BeginInteractivity();
+
+            // Wait for elements to be recreated with internal ids to permit mutation
+            WaitForNewElement(ref metaWithBindings, "meta-with-bindings");
+            WaitForNewElement(ref metaNoBindings, "meta-no-bindings");
+
+            // Validate updated head after prerender
+            Browser.Equal("Initial title", () => Browser.Title);
+            Browser.Equal("Initial meta content", () => metaWithBindings.GetAttribute("content"));
+            Browser.Equal("Immutable meta content", () => metaNoBindings.GetAttribute("content"));
+
+            // Change parameter of meta component
+            var inputMetaBinding = Browser.FindElement(By.Id("input-meta-binding"));
+            inputMetaBinding.Clear();
+            inputMetaBinding.SendKeys("Updated meta content\n");
+
+            // Wait for meta tag to be recreated with new attributes
+            WaitForNewElement(ref metaWithBindings, "meta-with-bindings");
+
+            // Validate new meta content attribute
+            Browser.Equal("Updated meta content", () => metaWithBindings.GetAttribute("content"));
+
+            void WaitForNewElement(ref IWebElement existingElement, string id)
```

Comment:
It's more common to use `existingElement = WaitForNewElement(existingElement, id)`, `ref` is most commonly used when dealing with value types